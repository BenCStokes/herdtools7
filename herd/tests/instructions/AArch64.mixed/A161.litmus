AArch64 A161
(* Test load exclusive; Store exclusive, symbolic location *)
(* Equivalent to A43 but testing mixed mode *)

Variant=mixed

 { 0:X4=z; uint64_t z=2; }

P0;
  LDXR X0, [X4];
  ADD X0, X0, #1;
  STXR W3, X0, [X4];

forall ((z=2 /\ 0:X3=1) \/ (z=3 /\ 0:X3=0))
