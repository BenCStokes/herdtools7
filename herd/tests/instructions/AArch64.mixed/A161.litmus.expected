Test A161 Required
States 2
0:X3=0x0; [z]=0x3;
0:X3=0x1; [z]=0x2;
Ok
Witnesses
Positive: 2 Negative: 0
Condition forall ([z]=0x2 /\ 0:X3=0x1 \/ [z]=0x3 /\ 0:X3=0x0)
Observation A161 Always 2 0
Hash=801cdab07257d3e78999199f60244eb9

